### PR TITLE
Add  `system-has-id` constraint

### PR DESCRIPTION
**Pull request by Gabeblis**
# Committer Notes
### Description
Added `system-characteristics` `has-system-id` constraint and tests for this constraint. These changes should be added as they are part of the effort write all of the constraints from the constraint tracker.

### Changes
- Added constraint `has-system-id` to fedramp-external-constraints.xml.
- Added pass and fail yaml tests for the above constraint.
- Added `ssp-has-system-id-INVALID.xml` file for this constraint's fail case.
### All Submissions:

- [x] Have you selected the correct base branch per [Contributing](https://github.com/GSA/fedramp-automation/blob/master/CONTRIBUTING.md) guidance?
- [x] Have you set "[Allow edits and access to secrets by maintainers](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork)"?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/GSA/fedramp-automation/pulls) for the same update/change?
- [x] Have you squashed any non-relevant commits and commit messages? \[[instructions](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History)\]
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
~~- [ ] If applicable, have all [FedRAMP Documents Related to OSCAL Adoption](https://github.com/GSA/fedramp-automation) affected by the changes in this issue have been updated.?~~
~~- [ ] If applicable, does this PR reference the issue it addresses and explain how it addresses the issue?~~

By submitting a pull request, you are agreeing to provide this contribution under the [CC0 1.0 Universal public domain](https://creativecommons.org/publicdomain/zero/1.0/) dedication.

### CODE REVIEW

File: src/validations/constraints/fedramp-external-constraints.xml [hunk 1]
```diff
@@ -137,6 +137,9 @@
             <expect id="has-data-flow-diagram-link-rel-allowed-value" target="system-characteristics/data-flow/diagram/link" test="@rel eq 'diagram'" level="ERROR">
                 <message>Each FedRAMP SSP data flow diagram must have a link rel attribute with the value "diagram".</message>
             </expect>
+            <expect id="has-system-id" target="system-characteristics" test="system-id[@identifier-type eq 'https://fedramp.gov']" level="ERROR">
+                <message>A FedRAMP SSP must have a FedRAMP system identifier.</message>
+            </expect>
```
Comment (aj-stein-gsa) [inline src/validations/constraints/fedramp-external-constraints.xml:142]: Do we need to explicitly create the style guide in #675 or can we begin recommending the prop as discussed for current and future PRs?

```suggestion
            <expect id="has-system-id" target="system-characteristics" test="system-id[@identifier-type eq 'https://fedramp.gov']" level="ERROR">
                <prop namespace="https://docs.oasis-open.org/sarif/sarif/v2.1.0" name="help-url" value="https://automate.fedramp.gov/documentation/ssp/4-ssp-template-to-oscal-mapping/#system-name-abbreviation-and-fedramp-unique-identifier"/>
                <message>A FedRAMP SSP must have a FedRAMP system identifier.</message>
            </expect>
```

Comment (Gabeblis) [reply src/validations/constraints/fedramp-external-constraints.xml:142]: > Do we need to explicitly create the style guide in #675 or can we begin recommending the prop as discussed for current and future PRs?

Are we officially moving forward style proposed in #735?

Comment (aj-stein-gsa) [reply src/validations/constraints/fedramp-external-constraints.xml:142]: Per today's conversation in standup, informal vote (from you, @wandmagic, and yours truly) is to notionally move forward. Sorry I took so long in the standup to get to a temp check on this.

Comment (Gabeblis) [reply src/validations/constraints/fedramp-external-constraints.xml]: done fbb783cc7e06cf730b46a8a13205f81640e78d29.

**Review by aj-stein-gsa — APPROVED** [review]

**Pull request merged**